### DISCUSSION TITLE
How To - start a Freetz-ng from scratch - on Windows - for the rest of us

### DISCUSSION THREAD

**Discussion by JohnPaden** [Freetz-NG/freetz-ng#165] [Allgemein]
This is meant to be as kiss as possible. For literally every step there are alternatives (Windows/Linux, VMWare/hyper-v/VirtualBox etc.). Example is a Fritzbox 7580.

None of the mentioned below is my work. Kudus and creds go to all that have brought Freetz and Freetz-ng to this point. Clearly there is a how to here https://github.com/Freetz-NG/freetz-ng where I learned a lot, too.
I am just summarizing what I did and what worked for me, because often the easy thing is the complex for many.

0) install the VM Environment of your choice (here VMWare Workstation Player)

1) download Freetz Linux Download from 
https://freetz.digital-eliteboard.com/?dir=Teamserver/Freetz/Freetz-VM/VirtualBox/Freetz-Linux-1.5.8

Start the OVA in your VMware, check if you have a ip adress (ifconfig).
I needed to 

sudo nano /etc/netplan/xxx.yaml

change the name of the network adapter to ens33

sudo netplan apply

2) Install freetz-ng

mdkir freetz-ng
git clone https://github.com/Freetz-NG/freetz-ng ~/freetz-ng

cd freetz-ng
sudo git pull
make menuconfig
- recommended for V7.2x due to current bug (01/2021: Other Patches - Disable AVM watchdog)
save config here
make

Now the image is in /freetz-ng/images/

3) build a in.memory image (needed for first installation of freetz-ng)
./freetz-ng/tools/image2inmemory ./freetz-ng/images/7580_07.21.xxxxxx.image ./freetz-ng/images/7580_07.21.xxxxxx.image.in.memory

4) Open the Fritzbox 7580 for Freetz

4.1) set the ip address of the windows network adapter to IP: 192.168.178.100 / Subnet: 255.255.255.0 / Gateway: 192.168.178.1 / DNS: 192.168.178.1

4.2.) Make a space to work - maybe on a seperate Windows computer.
md c:\patch
copy the ./freetz-ng/images/7580_07.21.xxxxxx.image and the ./freetz-ng/images/7580_07.21.xxxxxx.image.in.memory to c:\patch
copy the ./eva_tools/EVA-Discover.ps1 and ./eva_tools/EVA-FTP-Client.ps1 to c:\patch
(One easy way is via ftp in Total Commander)

4.3.) Open a Windows-Powershell as an Admin öffnen,
Set-ExecutionPolicy -Scope CurrentUser -ExecutionPolicy Unrestricted 
pull out the power plug and all but one LAN cables of the Fritzbox conneted directly to the PC

cd c:\Patch
.\EVA-Discover.ps1

plug in the power to the Fritzbox and wait
After less than a minute you should see
EVA_IP=192.168.178.1
True

.\EVA-FTP-Client.ps1 -ScriptBlock { BootDeviceFromImage -filename .\7580_07.21.xxxxxx.image.in.memory }

Fritzbox should blink and restart after several minutes. Be patient. 

To set the Execution back to normal:
Set-ExecutionPolicy -Scope CurrentUser -ExecutionPolicy Restricted

5.) Set up Fritzbox - to make sure both firmware partitions are loaded the same
default login (PW see back of the box)
stop the initial asistant
extended view
Freetz login admin / freetz

System Firmwareupdate
-> Freetz 7.21 mit Watchdox Fix einspielen -> reboot
System Firmwareupdate 2ter slot: 
Firmwarepartition wechseln
-> Freetz 7.21 mit Watchdox Fix einspielen -> reboot
System same two freetz-ng in both partitions

done.

**Comment by fda77** [comment]
What about if you write a .md (markdown) file like https://freetz-ng.github.io/freetz-ng/wiki/ ? MDs could be edited with a text editor and viewed with some tools oder browser plugins, i'm using https://addons.mozilla.org/de/firefox/addon/markdown-viewer-webext/
